docker/Dockerfile

```diff
@@ -44,14 +46,23 @@ RUN --mount=type=cache,target=/var/cache/sccache \
     just notice-file && \
     mv target/$(just arch=$TARGETARCH libc=gnu print-target)/release/restate-server target/restate-server
 
+FROM builder AS upload-false
+
+FROM builder AS upload-true
+RUN $(just --set arch $TARGETARCH --evaluate _arch)-linux-gnu-objcopy --only-keep-debug target/restate-server target/restate-server.debug
+RUN $(just --set arch $TARGETARCH --evaluate _arch)-linux-gnu-objcopy  --strip-debug target/restate-server
```

Comment:
I wonder if we should make stripping optional, or keep an option to copy debug symbols into the final container?

Reply:
imo there is no benefit to uploading the debug symbols if we aren't going to strip them. otherwise you can just run your tool against the binary directly? hence i treated the request to upload as a request to strip. nb we have a separate mechanism in the ci to keep debug symbols *but not upload*